File: Byte&Buy.AngularSPA/src/app/core/dto/offers/rent/rent-browser-item-response.ts
import { MoneyDto } from "../../common/money-dto";
import { OfferBrowserItemResponse } from "../common/offer-browser-item-response";

export interface RentBrowserItemResponse extends OfferBrowserItemResponse{
    type: 'rent';
    pricePerDay: MoneyDto;
    maxRentalDays: number;
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/RentOffer/RentBrowserItemResponse.cs
using ByteBuy.Core.DTO.Public.Money;
using ByteBuy.Core.DTO.Public.Offer.Common;

namespace ByteBuy.Core.DTO.Public.Offer.RentOffer;

//Dto for rent offer used in offer browser
public record RentBrowserItemResponse : OfferBrowserItemResponse
{
    public int MaxRentalDays { get; init; }
    public MoneyDto PricePerDay { get; init; } = null!;
}
